src/qnvimcore.cpp

```diff
@@ -148,13 +137,12 @@ autocmd VimEnter * let $MYQVIMRC=substitute($MYVIMRC, 'init.vim$', 'qnvim.vim',
         mNVim->api2()->nvim_subscribe("Gui");
         mNVim->api2()->nvim_subscribe("api-buffer-updates");
     });
+
+    m_cmdLine = std::make_unique<CmdLine>(this);
```

Comment:
Since it's a QObject, Isn't `new CmdLine(this)` enough to take care of garbage collection? I know using `new` is not preferred in C++ but last time I checked (I may not be updated) for QObjects you usually don't need to take care of garbage collection and passing `this` as the parent in the constructor is enough for Qt to take care of the rest.
I don't have a strong opinion about this, if you are confident this is better we can go with it, I'd be thankful if you provide me a resource so that I can learn more about it.

Reply:
I'm just trying to avoid standard library in parallel with Qt, my understanding is that Qt already has whatever standard library can provide for general use cases and having both work in parallel may cause some compatibility issues.